Microverse's Using Bootstrap project

sipe-daniel opened the pull request:
This project consists of building a Bootstrap document that matches the appearance of Newsweek.

sipe-daniel committed:
first commit in this branch

sipe-daniel committed:
perf(principal page): improve header main and footer

sipe-daniel committed:
perf(principal-page): improve the layout of the site components such header, nav, main, footer

christianotieno reviewed (changes requested):
# **CHANGES REQUIRED** :warning:
Hey @sipe-daniel  :wave: Good to see you making progress in the curriculum.
-
You have done a good job so far! 
We, however, do have some few changes to carry out before we proceed (see review comment).
##
For this project, we need to ensure nothing's messed up. Kindly see the difference between the clone and the source page below 👇 

|Source page|Clone Page|
|:---:|:---:|
|![source](https://user-images.githubusercontent.com/22345961/83085631-2e904700-a095-11ea-8497-e8f43a3924de.png)|![clone](https://user-images.githubusercontent.com/22345961/83085642-3354fb00-a095-11ea-87e2-c92db1425f4b.png)|

- [x] The contents of the cloned page are not properly centred as expected. Since this is a bootstraps project, we need to take full advantage of using its classes to style our page. A good fix for this case would be the use of the [`container`](https://www.w3schools.com/bootstrap4/bootstrap_containers.asp), class.
-  [x] Kindly see the difference in styles between the in the magazine sections of both pages below, and improve on it.

|Source Page|Clone Page|
|:---:|:--:|
|![in-the-magazine](https://user-images.githubusercontent.com/22345961/83085918-fb01ec80-a095-11ea-8ce0-d6e5eebeae54.png)|![in-the-magazine-clone](https://user-images.githubusercontent.com/22345961/83085927-005f3700-a096-11ea-8e5f-a17f31eb472c.png)|

- [x] Some more work needs to be carried out in the footer section as well. See the difference below. 👇 

|Source Page|Clone Page|
|:---:|:---:|
|![footer](https://user-images.githubusercontent.com/22345961/83086032-4ddba400-a096-11ea-887d-de552925e172.png)|![footer-clone](https://user-images.githubusercontent.com/22345961/83086039-52a05800-a096-11ea-8b48-437cd47e5baa.png)|

- [x] During Screen breakpoints, the small screen should only render one article per row, See the difference below and work on the changes. 

|Source Page|Clone Page|
|:---:|:---:|
|![small-screen](https://user-images.githubusercontent.com/22345961/83086222-d2c6bd80-a096-11ea-8889-5f6edbba0338.png)|![small-screen-clone](https://user-images.githubusercontent.com/22345961/83086231-d6f2db00-a096-11ea-809f-9d73edeee795.png)|

- [x]  The project should have a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] Make sure that with your next commits, that the Pull Request has a meaningful title and a short description ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

- [x] Commits should be very descriptive ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_commits.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)).
- [x] Enable sticler linter for this repo.
- [x] Make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright)
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

sipe-daniel commented:
Hi! Please this website change every day I started this project 2week ago.
Watch the project before giving comments. 
it is not normal to resign yourself before commenting on the project

sipe-daniel commented:
Hi! Please this website changes every day I started this project 2weeks ago.
![Capture d’écran_2020-05-28_09-06-59](https://user-images.githubusercontent.com/35634237/83117079-87c39f00-a0c4-11ea-94e0-be7b2858aec1.png)
![Capture d’écran_2020-05-28_09-06-40](https://user-images.githubusercontent.com/35634237/83117108-9316ca80-a0c4-11ea-847f-d77fd84989c5.png)

Watch the project before giving comments.
it is not normal, you have to inform you before commenting on the project. You can't compare the structure of my project with what is online now. The are many blocs which aren't on this website now, the structure has changed. You should ask yourself why my project is like that.

sipe-daniel commented:
![Capture d’écran_2020-05-28_09-15-31](https://user-images.githubusercontent.com/35634237/83117280-c8231d00-a0c4-11ea-916d-a18eb95114bf.png)
this part existed last week and doesn't exist this week

sipe-daniel commented:
![Capture d’écran_2020-05-28_09-26-06](https://user-images.githubusercontent.com/35634237/83118014-b4c48180-a0c5-11ea-9d89-204e41dae356.png)
![Capture d’écran_2020-05-28_09-25-22](https://user-images.githubusercontent.com/35634237/83118017-b5f5ae80-a0c5-11ea-9012-5eba43d296e7.png)
![Capture d’écran_2020-05-28_09-25-00](https://user-images.githubusercontent.com/35634237/83118020-b7bf7200-a0c5-11ea-8ec5-30c062627098.png)
![Capture d’écran_2020-05-28_09-24-09](https://user-images.githubusercontent.com/35634237/83118023-b8580880-a0c5-11ea-8b88-16553faace8b.png)

It's the same thing concerning this part.

sipe-daniel commented:
Thanks for your comments, I agree with the rest of your comments but concerning the structure,  I don't agree with that totally.

sipe-daniel commented:
Concerning your comments "**During Screen breakpoints, the small screen should only render one article per row, See the difference below and work on the changes.**" my project is ok I don't know why you say that. You can still check my project.
Thanks for your comments I'm going to work in order to improve my project.

sipe-daniel commented:
Please this website changes every day or week and I started this project 2weeks ago. so don't compare exactly all element in my project with the original page which is online now.

sipe-daniel committed:
refactor(principal-page) add readme and resolve some problems with css

concerning sm, xs, md, lg

sipe-daniel committed:
add an image(preview.png) in order to preview the project

sipe-daniel committed:
change the preview image

sipe-daniel committed:
change the preview image

lucasmazo32 commented on index.html:
```diff
@@ -0,0 +1,1049 @@
+          <div class="row justify-content-center">
+            <div class="col-lg-11 col-md-12 myclass justify-content-center">
+              <div class="row">
+                <div class="col-12  ml-lg-auto myclass">
+                  <ul class="nav-link">
+                    <li><a href="#">U.S.</a></li>
+                    <li><a href="#">World</a></li>
+                    <li><a href="#">Business</a></li>
+                    <li><a href="#">Tech & Science</a></li>
+                    <li><a href="#">Culture</a></li>
+                    <li><a href="#">Newsgeek</a></li>
+                    <li><a href="#">Sports</a></li>
+                    <li><a href="#">Health</a></li>
+                    <li><a href="#">The Debate</a></li>
+                    <li><a href="#">Vantage</a></li>
+                    <li><a href="#">Weather</a></li>
+                    <li>
+                        <div class="input-group">
+                            <input class="form-control py-2" type="search" value="search">
+                            <div class="input-group-text bg-transparent border-0"><i class="fa fa-search"></i></div>
+                        </div>
+                    </li>
+                  </ul>
```
- [x] The links and the input field appear in different columns. Kindly fix that.

![image](https://user-images.githubusercontent.com/57414789/83208878-7ad29a00-a11c-11ea-8f30-ce80d2a646d8.png)

sipe-daniel replied:
Thanks for your remarks. I rectified this, so it's already ok.

lucasmazo32 commented on index.html:
```diff
@@ -0,0 +1,1049 @@
+<!DOCTYPE  html> 
```
- [x] The layout in the Heroes of the pandemic section looks different. I am focusing on the way the images have different sizes. Kindly fix that.

Source Page | Clone Page
-- | --
![image](https://user-images.githubusercontent.com/57414789/83209159-3f849b00-a11d-11ea-8039-2768837bca81.png)  | ![image](https://user-images.githubusercontent.com/57414789/83209132-27ad1700-a11d-11ea-9f59-34d2eef2b46f.png)

lucasmazo32 commented on index.html:
```diff
@@ -0,0 +1,1049 @@
+<!DOCTYPE  html> 
```
- [x] We understand that the webpages change very often. One suggestion is to try to archive the webpage or take some screenshots for us to compere. There are details that always stay the same as the display of the editor's pick logo and text, the way that "see all features" in the first article is slightly bigger than the other pictures. I am going to point out those particular details here. Kindly fix the title layout and the first picture layout (I am just referring to image size).

Source | Clone
-- | --
![image](https://user-images.githubusercontent.com/57414789/83209815-d9991300-a11e-11ea-96c8-66104aea8194.png) | ![image](https://user-images.githubusercontent.com/57414789/83209776-be2e0800-a11e-11ea-9731-a1b44907ae0d.png)

sipe-daniel replied:
Thanks for your comments. They helped me a lot.

lucasmazo32 reviewed (changes requested):
# Changes required :wrench:

You are doing a great job so far, and the project is looking good. 

For the project to get approved, kindly implement the changes mentioned below.

### Happy coding! :computer:

## Lucas Mazo

sumancrest0001 reviewed (changes requested):
**Status: Changes required 
Hello @sipe-daniel,**
You have done a good job so far and the project looks beautiful. But we need to make a couple of changes to get approval. I did not see any changes in the layout and it still has all the issues stated by the previous TSE. I checked your project on githack as well as on chrome. Let me know if you have a different layout. In addition to that, we have to fix following issues also.

[NOTE] Optional changes are not mandatory but nice to have.

[Required] we need to make the signup form responsive on a medium screen. On the small screen, subscribe and newsletter section looks different from the original one. Please make the required changes.

[optional] Let's give similar font-size, font-width, and color to titles as it is a clone project.

**Kindly make the required changes and request for another review.**
if you have any query, feel free to contact me on slack **Suman Shrestha**

All the best with the next submission.
**Happy coding !!!**

sipe-daniel committed:
refactor(principal-page):
improve the responsive of the page, change the title font-family

about the responsive I worked on signup, subscribe and newsletter section. There were some problems on medium screen concerning the signup part and on small screen for subscribe and newsletter part

sipe-daniel committed:
refactor(principal-page): center the image inside the subscribe section

I centred this image for a small screens

sipe-daniel committed:
refactor(principal-page): change the style of some titles

I changed the colors, the font-family and the size of some titles and subtitles

sumancrest0001 reviewed (changes requested):
**STATUS: CHANGES REQUIRED
Excellent work @sipe-daniel :clap: :+1:** 

Thank you for making the requested changes and the project looks very close to the original and breaking points are working correctly. But We are missing two main points. 

[REQUIRED] The hamburger menu is missing for the small and medium screens.
[REQUIRED] Social media icons are missing from the footer section. and make it responsive.
Currently, on a small screen.  :point_down: 
![Screenshot from 2020-05-30 15-21-23](https://user-images.githubusercontent.com/37747619/83325081-56e09700-a289-11ea-89a0-80ffd16b8ecc.png)

sipe-daniel committed:
add icones

sipe-daniel committed:
refactor(principal-page): improve reponsive icones

sumancrest0001 reviewed (approved):
**STATUS: APPROVED :fire: :fire: 
Awesome work @sipe-daniel,**
You have ticked all the requirements. the layout and responsiveness are very close to the original page. Now, you can merge the working branch to master and move to the next project.

**Happy coding !!!**

sipe-daniel committed:
refactor(principal-page): vertical centering of the elements

I centered vertically the text IN THE MAGAZINE
